### PR TITLE
Fix: `no-useless-rename` false positive in babel-eslint (fixes #6266)

### LINKED ISSUE #6266: no-useless-rename triggered when there's no rename
**What version of ESLint are you using?**
v2.11.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Default (babel-eslint shows the same issue)

**Please show your full configuration:**
```json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "rules": {
    "no-useless-rename": 1
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
import Foo from 'foo';
import {bar as otherBar} from 'bar';
import {fooBar} from 'foo-bar';
import {baz as baz} from 'baz';
```

**What did you expect to happen?**
Just one warning in line 4.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/path-to-the-file/index.js
  3:9  warning  Import fooBar unnecessarily renamed  no-useless-rename
  4:9  warning  Import baz unnecessarily renamed     no-useless-rename

✖ 2 problems (0 errors, 2 warnings)
```

### PR DESCRIPTION
**Pull request by alberto**
Sorry, this one should fix it for real. I manually tested it using `babel-eslint`.

### CODE REVIEW

**Comment by platinumazure** [comment]
How does performance look, out of curiosity?

File: lib/rules/no-useless-rename.js [hunk 1]
```diff
@@ -109,7 +115,7 @@ module.exports = {
             }
 
             if (node.imported.name === node.local.name &&
-                    node.imported !== node.local) {
+                    !lodash.isEqual(node.imported, node.local)) {
```
Comment (nzakas) [inline lib/rules/no-useless-rename.js]: Can't you just check that `range[0]` isn't the same? We really only care about the location, right?

Comment (alberto) [inline lib/rules/no-useless-rename.js]: Yeah, that would be better, thanks!

**Comment by nzakas** [comment]
Lgtm

**Comment by kaicataldo** [comment]
Sorry, thought this was my branch - restored it!

**Pull request merged**